src/promptflow-tracing/promptflow/tracing/_trace.py

```diff
@@ -455,7 +435,9 @@ def wrapped(*args, **kwargs):
         span_name = get_node_name_from_context(used_for_span_name=True) or trace.name
         # need to get everytime to ensure tracer is latest
         otel_tracer = otel_trace.get_tracer("promptflow")
-        with otel_tracer.start_as_current_span(span_name) as span, _record_cancellation_exceptions_to_span(span):
+        with otel_tracer.start_as_current_span(
```

Comment:
Maybe we can have our own `start_as_current_span` to avoid duplicated if/end logic in multiple places?

Reply:
Sure. Will do